### DISCUSSION TITLE
WorkerService .NET Core 6 Screenshot in Linux ARM64.

### DISCUSSION THREAD

**Discussion by thanhthuongpham** [dlemstra/Magick.NET#1478] [Help]
Hello,
I try to Screenshot then save them to file. 

```
      MagickReadSettings settings = new MagickReadSettings()
      {
          ExtractArea = new MagickGeometry(0, 0, 800, 480)
      };
      MagickImage image = new MagickImage("screenshot:", settings);
      image.Read("screenshot:", settings);
      byte[] data = image.ToByteArray(MagickFormat.Jpg);
      File.WriteAllBytes($"/home/pi/Pictures/{DateTime.Now.ToString("yyyy_MM_dd_HH_mm_ss")}.jpg", data);
```

//OR
```
      using (MagickImage mImage = new MagickImage("screenshot:",800,480))
      {
          byte[] dataU = mImage.ToByteArray(MagickFormat.Jpg);
          File.WriteAllBytes($"/home/pi/Pictures/{DateTime.Now.ToString("yyyy_MM_dd_HH_mm_ss")}.jpg", dataU);
      }
```
But I have been an error:
The type initializer for 'NativeMagickSettings' threw an exception.
at ImageMagick.MagickSettings.NativeMagickSettings..ctor() in /_/src/Magick.NET/Native/Settings/MagickSettings.cs:line 343
at ImageMagick.MagickSettings..ctor() in /_/src/Magick.NET/Settings/MagickSettings.cs:line 33

---
I'm coding a WorkerService .NET Core 6 and deploy as Systemd in Raspberry Pi 4 Model B (OS : Debian GNU/Linux 12 (bookworm))
I'm using Magick.NET-Q16-arm64 (13.4.0)
Please help me solve this problem.
Thanks!

**Comment by dlemstra** [comment]
Does it not tell you more than `The type initializer for 'NativeMagickSettings' threw an exception.`? Not sure what is  happening but I suspect the `Magick.Native-` library is not in your bin folder?

**Reply by thanhthuongpham** [reply]
Hi dlemstra,
Thanks you for your advice.
After copy the Magick.Native- , this issue has been fixed. But I have been new issue when I try Screenshot:
```
using (MagickImage mImage = new MagickImage("screenshot:"))
{
    byte[] data = mImage.ToByteArray(MagickFormat.Jpeg);
    File.WriteAllBytes($"/home/pi/Pictures/{DateTime.Now.ToString("yyyy_MM_dd_HH_mm_ss")}.jpeg", data);
}
```
this code threw exception: NegativeOrZeroImageSize, because the mImage after line 'using (MagickImage mImage = new MagickImage("screenshot:"))' is no data, I have checked mImage.Width is 0 and mImage.Height is 0. 
Am I mistake somewhere?

**Reply by dlemstra** [reply]
It looks like ImageMagick only supports this when it's build with the X11 delegate and that is not happening for the non-Windows build of Magick.NET. And I am also not sure if there are any alternative libraries that you can use for this.

**Reply by thanhthuongpham** [reply]
I guess I have to research another way. Thank you very much!
